source/common/http/filter_manager.cc

```diff
@@ -407,6 +407,10 @@ ResponseHeaderMapOptRef ActiveStreamDecoderFilter::continueHeaders() const {
 
 void ActiveStreamDecoderFilter::encodeHeaders(ResponseHeaderMapPtr&& headers, bool end_stream,
                                               absl::string_view details) {
+  const absl::optional<int64_t> status = Utility::getResponseStatusNoThrow(*headers);
+  if (status.has_value()) {
+    parent_.stream_info_.setResponseCode(status.value());
```

Comment:
As far as I know, this hasn't been set anywhere for normal request -- only set for local replies.

Reply:
Ok I see, thanks for looking into that.

I wonder if we want to do more here to ensure that it reflects the final downstream code. With this PR we'd set it before the filters see it (which is probably what you want?) but it would be inaccurate if any of the filters modify the status code. I wonder if there's a way to make it more consistent? Should this just read it from the response headers instead of being a separate value on the stream info?

@alyssawilk maybe you have thoughts here?

Reply:
> I wonder if we want to do more here to ensure that it reflects the final downstream code. With this PR we'd set it before the filters see it (which is probably what you want?) but it would be inaccurate if any of the filters modify the status code. I wonder if there's a way to make it more consistent? Should this just read it from the response headers instead of being a separate value on the stream info? 

yeah that's a good point I was also thinking about. The original issue #18783 is basically requiring the "current" status code at the time when Wasm extensions try to retrieve the status code via stream_info. So maybe we should set this anytime when the :status header value has changed. But I think this also applies to responseCodeDetail.

Reply:
OK I think this complicates the problem -- maybe we should change the implementation of CEL wrapper so that it uses the value of :status directly instead of stream_info->responseCode().